src/kademlia/traversal_algorithm.cpp

```diff
@@ -118,85 +140,110 @@ void traversal_algorithm::add_entry(node_id const& id
 		done();
 		return;
 	}
+
+	o->flags |= flags;
+
 	if (id.is_all_zeros())
 	{
 		o->set_id(generate_random_id());
 		o->flags |= observer::flag_no_id;
-	}
 
-	o->flags |= flags;
+		m_results.push_back(o);
```

Comment:
I just push back here, causing most of this function to end up in the "else" block, and gaining one level of indentation (which makes the patch look ugly)